src/LanguageServer/Protocol/Protocol/TextDocumentContentChangeEvent.cs

```diff
@@ -18,6 +18,7 @@ internal sealed class TextDocumentContentChangeEvent
     /// Gets or sets the range of the text that was changed.
     /// </summary>
     [JsonPropertyName("range")]
+    [JsonRequired]
     public Range Range
```

Comment:
Rather than messing with the SumTypes, could we have just made this nullable?

Reply:
The spec [defines it as a union type](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentContentChangeEvent), so we should use a union type.  They do happen to share the same field names right now, but that isn't a guarantee for the futre